HIP/hip-904-frictionless-airdrop.md

```diff
@@ -0,0 +1,736 @@
+
+If the receiver’s account is **not** associated with the token, and they have **no** free auto-association slots, then the airdrop transfer neither succeeds nor fails.
+
+In this case a *pending transfer* is created and held in state. Tokens in the pending transfer are not deducted from the sender, nor are they credited to the receiver, until the receiver *claims* the airdrop. If the receiver never claims the airdrop, then the sender continues to pay for it forever, or until their account expires, is deleted, or the sender *cancels* the pending transfer.
+
+All pending transfers sent by an account must be canceled before the account can be deleted.
+
+At the time a token transfer is stored as a *pending transfer*, any custom fees are assessed and stored. If a token A has custom fees of 1 HBAR, and the sender sends that token, the fee of 1 HBAR is assessed at the time of the TokenAirdrop transaction. Even if the custom fee is changed to 100 HBAR between the time the transaction is stored and claimed, no additional custom fee will be assessed to the sender. Additionally, no refund of custom fees from the network to the payer will be provided should the sender cancel the pending transfer.
+
+When the sender’s account is renewed, each pending transfer owned by the sender is taken into account and paid for as part of the renewal process. This is in addition to the regular rent due as a result of the token balance owned by the sender. If the sender’s account expires and cannot be renewed, then all pending transfers for that sender are canceled. Any attempt to claim a missing or canceled pending transfer, or a pending transfer from an expired or deleted sender account, will fail. Conceptually, each pending transaction is immediately canceled at the time the account expired.
+
+### Token Claim Airdrop
+
+To support a users ability to claim a pending transfer a new `TokenClaimAirdrop` transaction will be introduced.
+When a receiver claims a token, the `TokenClaimAirdrop` transaction **must** be signed by the receiver, in addition to the transaction payer (who may be the same as the receiver). In essence, if an account has no free auto-association slots, then it behaves as though receiver-sig-required were true *for airdrops*. That is, the receiver must sign the transaction to receive the airdrop.
+
```

Comment:
Do we want to add a note about how this is expected to work re wallets like we have described in the section below?

It is expected that wallets would enable user flows to claim their pending transfers and facilitate creating, signing, and submission of the transaction.

Reply:
Yes, there's a section for wallets at the end with suggestions of features they should expose.